modules/core/arrow-test/src/main/kotlin/arrow/test/laws/MonadDeferLaws.kt

```diff
@@ -14,18 +14,18 @@ import kotlinx.coroutines.experimental.CommonPool
 import kotlinx.coroutines.experimental.newSingleThreadContext
 
 object MonadDeferLaws {
-  inline fun <F> laws(SC: MonadDefer<F>, EQ: Eq<Kind<F, Int>>, EQ_EITHER: Eq<Kind<F, Either<Throwable, Int>>>, EQERR: Eq<Kind<F, Int>> = EQ): List<Law> =
-    MonadErrorLaws.laws(SC, EQERR, EQ_EITHER, EQ) + listOf(
+  inline fun <F> laws(SC: MonadDefer<F>, EQ: Eq<Kind<F, Int>>, EQ_EITHER: Eq<Kind<F, Either<Throwable, Int>>>): List<Law> =
+    BracketLaws.laws(SC, EQ, EQ_EITHER, EQ) + listOf(
       Law("Sync bind: binding blocks") { SC.asyncBind(EQ) },
-      Law("Sync bind: binding failure") { SC.asyncBindError(EQERR) },
+      Law("Sync bind: binding failure") { SC.asyncBindError(EQ) },
       Law("Sync bind: unsafe binding") { SC.asyncBindUnsafe(EQ) },
-      Law("Sync bind: unsafe binding failure") { SC.asyncBindUnsafeError(EQERR) },
+      Law("Sync bind: unsafe binding failure") { SC.asyncBindUnsafeError(EQ) },
       Law("Sync bind: binding in parallel") { SC.asyncParallelBind(EQ) },
       Law("Sync bind: binding cancellation before flatMap") { SC.asyncCancellationBefore(EQ) },
       Law("Sync bind: binding cancellation after flatMap") { SC.asyncCancellationAfter(EQ) },
       Law("Sync bind: bindingInContext cancellation before flatMap") { SC.inContextCancellationBefore(EQ) },
       Law("Sync bind: bindingInContext cancellation after flatMap") { SC.inContextCancellationAfter(EQ) },
-      Law("Sync bind: bindingInContext throw equivalent to raiseError") { SC.inContextErrorThrow(EQERR) },
+      Law("Sync bind: bindingInContext throw equivalent to raiseError") { SC.inContextErrorThrow(EQ) },
```

Comment:
Are you sure this needs to be changed?

Reply:
EQERR is an EQ instance for `Kind<F, Int>`, exactly the same than other param: `EQ`. If you look at callers, feels like we're passing the same instance twice everywhere:

![captura de pantalla 2018-11-10 a las 8 53 27](https://user-images.githubusercontent.com/6547526/48298950-23cb8c80-e4c6-11e8-8477-d1864597fa4f.png)

So I removed it since I didn't understand why we need two if they're just structural comparisons over the same `Kind<F, Int>`, so can't find a reason for them to be different. Please let me know if I'm wrong here. Actually I think I forgot to remove `EQERR` also in `AsyncLaws`, since it has the same issue.

Reply:
Thinking about this, there's a potential issue where one of the tests returns an unexpected error, and our Eq accepts it because it's made to compare errors. Same for error cases, they may succeed and the Eq treats them as correct.

This is probably an issue with the callers, more than it is with the laws.

Reply:
Alright, I can see the problem now. I've reverted it to be as it was til today, since it doesn't look like a boyscout we wanna do here, given the highlighted caveats of the approach. Still, it might be something to address in the maintainer's meeting.